src/sentry/services/hybrid_cloud/organization.py

```diff
@@ -12,36 +25,109 @@
 from sentry.silo import SiloMode
 
 
+@dataclass
+class ApiTeam:
+    id: int = -1
+    status: int = TeamStatus.VISIBLE
+    organization_id: int = -1
+    slug: str = ""
+
+
+@dataclass
+class ApiTeamMember:
+    id: int = -1
+    is_active: bool = False
+    role: Optional[TeamRole] = None
+    project_ids: List[int] = field(default_factory=list)
+    scopes: List[str] = field(default_factory=list)
+    team_id: int = -1
+
+
+@dataclass
+class ApiProject:
+    id: int = -1
+    slug: str = ""
+    name: str = ""
+    organization_id: int = -1
+    status: int = ProjectStatus.VISIBLE
+
+
 @dataclass
 class ApiOrganizationMember:
+    id: int = -1
+    organization_id: int = -1
     # This can be null when the user is deleted.
-    user_id: Optional[int]
-    pass
+    user_id: Optional[int] = None
+    member_teams: List[ApiTeamMember] = field(default_factory=list)
+    role: str = ""
+    project_ids: List[int] = field(default_factory=list)
+    scopes: List[str] = field(default_factory=list)
+
+
+@dataclass
+class ApiOrganizationFlags:
+    allow_joinleave: bool = False
+    enhanced_privacy: bool = False
+    disable_shared_issues: bool = False
+    early_adopter: bool = False
+    require_2fa: bool = False
+    disable_new_visibility_features: bool = False
+    require_email_verification: bool = False
 
 
 @dataclass
 class ApiOrganization:
     slug: str = ""
     id: int = -1
-    # exists iff the organization was queried with a user_id context, and that user_id
+    # exists if and only if the organization was queried with a user_id context, and that user_id
     # was confirmed to be a member.
+    # member: Optional[ApiOrganizationMember] = None
+    name: str = ""
+
+    # Represents the full set of teams and projects associated with the org.
+    teams: List[ApiTeam] = field(default_factory=list)
+    projects: List[ApiProject] = field(default_factory=list)
+
+    flags: ApiOrganizationFlags = field(default_factory=lambda: ApiOrganizationFlags())
+
+
+@dataclass
+class ApiUserOrganizationContext:
+    user_id: Optional[int]
+    organization: ApiOrganization = field(default_factory=lambda: ApiOrganization())
+    # Set iff the user_id in this object has a membership for the requested organization.
+    # Note that all related fields of this organization member are filtered by visibility and is_active=True.
     member: Optional[ApiOrganizationMember] = None
 
+    def __post_init__(self) -> None:
+        if self.user_id is not None and self.member is not None:
+            assert self.user_id == self.member.user_id
+
 
 class OrganizationService(InterfaceWithLifecycle):
+    @abstractmethod
+    def get_organization_by_id(
```

Comment:
Do we need `organization` in the method names if the service also has organization in the name?

Reply:
So the problem is that some organization service methods don't act on orgs, but say, on org membership.  As long as the service handles both membership and the orgs, I felt it more descriptive to include the type name.

I didn't think it was wise to separate org service from org member service.  This organization (haha) made more sense to me in the end.